src/DynamoCoreWpf/Utilities/NodeContextMenuBuilder.cs

```diff
@@ -0,0 +1,451 @@
+﻿using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using Dynamo.Controls;
+using Dynamo.Graph.Nodes;
+using Dynamo.ViewModels;
+
+namespace Dynamo.Wpf.Utilities
+{
+    /// <summary>
+    /// A static class containing methods for dynamically building a Node's Context Menu.
+    /// </summary>
+    internal static class NodeContextMenuBuilder
```

Comment:
1. please use nameof where you can, when these strings change, everything will break.
2. There is a lot of work in here to construct the views and bindings with code - thats fine - but I think the xaml could also have been reused by extracting it to a resource and instantiating it - perhaps that would not play well with the dynamic menu items - but I think if we can avoid re-writing all of our xaml we should.

Reply:
1. I've inserted nameof everywhere. The only hard-coded strings in the class are now equivalents of those that were previously hard-coded in XAML.
2. I'm afraid I don't understand this approach. Please feel free to reach out on Slack/Zoom if you want to go over this.